hopsworks-common/src/main/java/io/hops/hopsworks/common/dataset/FolderNameValidator.java

```diff
@@ -65,72 +55,41 @@
    * @param name
    * @param subdir Indicates a directory under a top-level dataset
    */
+  private static Pattern datasetNameRegex = Pattern.compile("^((?!__)[-_a-zA-Z0-9\\.]){1,87}[-_a-zA-Z0-9]$");
+  private static Pattern subDirNameRegex = Pattern.compile("^((?!__)[-_a-zA-Z0-9 \\.]){0,87}[-_a-zA-Z0-9]$");
+
   public static void isValidName(String name, boolean subdir) throws DatasetException {
-    String reason = "";
-    boolean valid = true;
-    if (name == null || name.isEmpty()) {
-      valid = false;
-      reason = ResponseMessages.FOLDER_NAME_NOT_SET;
-    } else if (name.length() > 88) {
-      valid = false;
-      reason = ResponseMessages.FOLDER_NAME_TOO_LONG;
-    } else if (name.endsWith(".")) {
-      valid = false;
-      reason = ResponseMessages.FOLDER_NAME_ENDS_WITH_DOT;
-    } else if (name.contains("" + Settings.DOUBLE_UNDERSCORE)) {
-      valid = false;
-      reason = ResponseMessages.FOLDER_NAME_CONTAIN_DISALLOWED_CHARS
-              + Settings.FILENAME_DISALLOWED_CHARS + Settings.DOUBLE_UNDERSCORE;
+    if (name == null) {
+      throw new DatasetException(RESTCodes.DatasetErrorCode.DATASET_NAME_INVALID, Level.FINE, "Empty name");
```

Comment:
@SirOibaf this error suggests to the client they invoked this method with `name` set as `Empty name`. Shouldn't we throw IllegalArgumentException here?

Reply:
Ok